medmodels/treatment_effect/continuous_estimators.py

```diff
@@ -153,37 +187,62 @@ def cohens_d(
     Raises:
         ValueError: If the outcome variable is not numeric.
     """
-    treated_outcomes = np.array(
-        [
-            medrecord.edge[
-                find_reference_edge(
-                    medrecord,
-                    node_index,
-                    outcome_group,
-                    time_attribute=time_attribute,
-                    reference=reference,
-                )
-            ][outcome_variable]
-            for node_index in treatment_outcome_true_set
-        ]
-    )
+    if time_attribute is not None:
+        treated_outcomes = np.array(
+            [
+                medrecord.edge[
+                    find_reference_edge(
+                        medrecord,
+                        node_index,
+                        outcome_group,
+                        time_attribute=time_attribute,
+                        reference=reference,
+                    )
+                ][outcome_variable]
+                for node_index in treatment_outcome_true_set
+            ]
+        )
+        control_outcomes = np.array(
+            [
+                medrecord.edge[
+                    find_reference_edge(
+                        medrecord,
+                        node_index,
+                        outcome_group,
+                        time_attribute="time",
+                        reference=reference,
+                    )
+                ][outcome_variable]
+                for node_index in control_outcome_true_set
+            ]
+        )
+    else:
+        edges_treated_outcomes = medrecord.select_edges(
+            lambda edge: query_edges_set_outcome(
+                edge, treatment_outcome_true_set, outcome_group
+            )
+        )
+        treated_outcomes = treated_outcomes = np.array(
+            [
+                medrecord.edge[edge_id][outcome_variable]
+                for edge_id in edges_treated_outcomes
+            ]
+        )
+
+        edges_control_outcomes = medrecord.select_edges(
+            lambda edge: query_edges_set_outcome(
+                edge, control_outcome_true_set, outcome_group
```

Comment:
I'm not 100% sure that I understand the use of this. If the set is already treatment/control and outcome_true, why are we checking again if they are connected to the outcome group? Or is this just the naming of the variables that is confusing to me?

Reply:
It is retrieving all edges that connect those nodes in the control outcome true set with the outcome. It is not checking whether they are connected, but retrieving the edges that connect them!